src/Compiler/Service/TransparentCompiler.fs

```diff
@@ -2243,13 +2298,31 @@ type internal TransparentCompiler
                             yield "--nowarn:" + code
                     ]
 
+                // Once we do have the script closure, we can populate the cache to re-use can later.
+                let loadClosureKey =
+                    mkScriptClosureCacheKey
+                        fileName
+                        sourceText
+                        useSimpleResolution
+                        useFsiAuxLib
+                        useSdkRefs
+                        assumeDotNetFramework
+                        (projectFileName, "")
+                        otherFlags
+                        optionsStamp
+
+                // Populate the cache.
+                let! _ =
+                    caches.ScriptClosure.Get(loadClosureKey, node { return loadClosure })
```

Comment:
This will only populate the cache if the key isn't already there, is that expected?

Reply:
Given the `ISourceTextNew` is part of the key, I would also expect this to more or less always be the case that the cache won't have the entry.

Unless `GetProjectSnapshotFromScript` is being called with the exact same input twice, I don't think this will ever happen.